guardrails/validator_base.py

```diff
@@ -243,6 +337,44 @@ def validate_stream(
             validation_result.validated_chunk = chunk_to_validate
         return validation_result
 
+    def _hub_inference_request(self, request_body: dict) -> Any:
+        """Makes a request to the Validator Hub to run a ML based validation model. This
+        request is authed through the hub and rerouted to a hosted ML model. The reply
+        from the hosted endpoint is returned and sent to this client.
+
+
+        Args:
+            request_body (dict): A dictionary containing the required info for the final
+            inference endpoint to run.
+
+        Raises:
+            HttpError: If the recieved reply was not ok.
+
+        Returns:
+            Any: Post request response from the ML based validation model.
+        """
+
+        try:
+            submission_url = self.validation_endpoint
+
+            headers = {
+                "Authorization": f"Bearer {self.hub_jwt_token}",
+                "Content-Type": "application/json",
+            }
+            req = requests.post(submission_url, json=request_body, headers=headers)
+
+            body = req.json()
+            if not req.ok:
```

Comment:
I think req.ok might have to happen before req.json().  req.json might raise an exception if the body isn't valid json.